lua/telescope/actions/init.lua

```diff
@@ -408,18 +408,50 @@ actions.send_to_qflist = function(prompt_bufnr)
 
   actions.close(prompt_bufnr)
 
-  vim.fn.setqflist(qf_entries, 'r')
+  vim.fn.setqflist(qf_entries, mode)
 end
 
-actions.smart_send_to_qflist = function(prompt_bufnr)
+actions.send_selected_to_qflist = function(prompt_bufnr)
+  send_selected_to_qf(prompt_bufnr, 'r')
+end
+
+actions.add_selected_to_qflist = function(prompt_bufnr)
+  send_selected_to_qf(prompt_bufnr, 'a')
+end
+
+actions.send_to_qflist = function(prompt_bufnr)
+  send_all_to_qf(prompt_bufnr, 'r')
+end
+
+actions.add_to_qflist = function(prompt_bufnr)
+  send_all_to_qf(prompt_bufnr, 'a')
+end
+
+local smart_send = function(prompt_bufnr, mode)
   local picker = action_state.get_current_picker(prompt_bufnr)
   if table.getn(picker:get_multi_selection()) > 0 then
-    actions.send_selected_to_qflist(prompt_bufnr)
+    if mode == 'r' then
+      actions.send_selected_to_qflist(prompt_bufnr)
+    else
+      actions.add_selected_to_qflist(prompt_bufnr)
+    end
```

Comment:
Can't we just use the other local function here and pass in `mode`

Reply:
oh yeah, we can. Don't know why I didn't thought of that.